packages/hooks/use-archive-infinite-scroll.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect } from "react";
+import { useConnect, connect, css } from "frontity";
+import { Connect } from "frontity/types";
+import useInfiniteScroll from "./use-infinite-scroll";
+import WpSource from "@frontity/wp-source/types";
+import TinyRouter from "@frontity/tiny-router/types";
+
+type Wrapper = React.FC<Connect<WpSource & TinyRouter, { link: string }>>;
```

Comment:
Are you using the types of `WpSource` and `TinyRouter` instead of `Source` and `Router` for something in particular?

Reply:
Just because I tend to use the types of the packages I'm working with. But in this case I guess we could use the generics packages, so different implementations are compatibles?

Reply:
Yes, it should work with any theme or source so it's better if we use the general ones.